### DISCUSSION TITLE
Format plot x-axis from timestamp to readable date

### DISCUSSION THREAD

**Discussion by lalamm** [emilk/egui#919] [Q&A]
Hello!

I've a timeseries and would like to show readable dates instead of timestamp. When i'm reading the documentation it says the following for [egui::widgets::plot::Value](https://docs.rs/egui/latest/egui/widgets/plot/struct.Value.html)

``
Uses f64 for improved accuracy to enable plotting large values (e.g. unix time on x axis).
``

But i'm unsure where to set the formatting behaviour.

Looks like this currently

![image](https://user-images.githubusercontent.com/13511791/143950933-4a766ad2-6446-410e-b4f3-709684936ffe.png)

Kind regards,
Ludvig

**Comment by emilk** [comment]
This was added in https://github.com/emilk/egui/pull/934

**Reply by MarkTurney** [reply]
Emil, I have just tried out custom_label_func() which allows us to customize the cursor label, but I believe that this question is in regards to the labels printed along the x-axis.  I am also interested in a way to plot timestamps in terms of date.

I have included an image to capture the question.  Is there currently a way to format the x-axis so that a timestamp displays as 2020-04-02 as it does in the below image?

![date_x_axis](https://user-images.githubusercontent.com/4324877/147686643-471b2520-9c6c-4a20-a566-0e6dba26456c.jpg)

**Reply by lalamm** [reply]
> Emil, I have just tried out custom_label_func() which allows us to customize the cursor label, but I believe that this question is in regards to the labels printed along the x-axis. I am also interested in a way to plot timestamps in terms of date.
> 
> I have included an image to capture the question. Is there currently a way to format the x-axis so that a timestamp displays as 2020-04-02 as it does in the below image?
> 
> ![date_x_axis](https://user-images.githubusercontent.com/4324877/147686643-471b2520-9c6c-4a20-a566-0e6dba26456c.jpg)

Yes! It is the X-axis formatting i want :)

**Reply by emilk** [reply]
Ah, no that is not yet possible. It should be fairly easy to add a custom callback for it though!

**Reply by MarkTurney** [reply]
Hello Emil, I am very interested in how to create this custom callback.  Could you point me to a similar example and exactly which API I would need to register the callback with?

Thanks a ton!

**Reply by emilk** [reply]
The method that paints the axes are called `paint_axis` in `egui/src/widgets/plot/mod.rs`. There is a line `let text = emath::round_to_decimals(value_main, 5).to_string(); // hack` that could be replaced by a call to a callback instead (`let text = text_from_value(value_main);` or something).
